src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimantResponseAgreedRepaymentRespondentNotificationHandler.java

```diff
@@ -0,0 +1,126 @@
+import uk.gov.hmcts.reform.civil.config.PinInPostConfiguration;
+import uk.gov.hmcts.reform.civil.config.properties.notification.NotificationsProperties;
+import uk.gov.hmcts.reform.civil.enums.YesOrNo;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.service.NotificationService;
+import uk.gov.hmcts.reform.civil.service.OrganisationService;
+import uk.gov.hmcts.reform.prd.model.Organisation;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.NOTIFY_RESPONDENT1_FOR_CLAIMANT_AGREED_REPAYMENT;
+import static uk.gov.hmcts.reform.civil.utils.PartyUtils.getPartyNameBasedOnType;
+
+@Service
+@RequiredArgsConstructor
+public class ClaimantResponseAgreedRepaymentRespondentNotificationHandler extends CallbackHandler implements NotificationData {
+
+    private static final List<CaseEvent> EVENTS = List.of(
+        NOTIFY_RESPONDENT1_FOR_CLAIMANT_AGREED_REPAYMENT);
+    public static final String TASK_ID = "ClaimantAgreedRepaymentNotifyRespondent1";
+    private static final String REFERENCE_TEMPLATE = "claimant-agree-repayment-respondent-notification-%s";
+
+    private final NotificationService notificationService;
+    private final NotificationsProperties notificationsProperties;
+    private final OrganisationService organisationService;
+    private final PinInPostConfiguration pipInPostConfiguration;
+
+    @Override
+    protected Map<String, Callback> callbacks() {
+        return Map.of(
+            callbackKey(ABOUT_TO_SUBMIT),
+            this::notifyRespondent1ForAgreedRepayment
+        );
+    }
+
+    @Override
+    public String camundaActivityId(CallbackParams callbackParams) {
+        return TASK_ID;
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse notifyRespondent1ForAgreedRepayment(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+
+        if ((YesOrNo.YES.equals(caseData.getRespondent1OrgRegistered())
+            && caseData.getRespondentSolicitor1EmailAddress() == null)
+            || ((!YesOrNo.YES.equals(caseData.getRespondent1OrgRegistered())
+            && caseData.getRespondent1().getPartyEmail() == null
+            ))) {
+            return AboutToStartOrSubmitCallbackResponse.builder().build();
+        }
+
+        notificationService.sendMail(
+            addEmail(caseData),
+            addTemplate(caseData),
+            addProperties(caseData),
+            String.format(REFERENCE_TEMPLATE, caseData.getLegacyCaseReference())
+        );
+        return AboutToStartOrSubmitCallbackResponse.builder().build();
+    }
+
+    @Override
+    public Map<String, String> addProperties(CaseData caseData) {
+        if (YesOrNo.YES.equals(caseData.getRespondent1OrgRegistered())) {
```

Comment:
because  you are repeating the code here with check YesOrNo.Yes etc in several places could you please put it in private boolean method and name is something meaningful like isRespondentRepresented() ?

Reply:
also as we discussed wouldn't it be better to check on the field that indicates that the respondent is represented or not